src/main.rs

```diff
@@ -164,33 +166,46 @@ async fn main() -> Result<()> {
                 inputs: Input::IpldData(ipld_args),
             };
 
-            let closure_ipld: Ipld = closure.into();
+            let closure_ipld: Ipld = closure.clone().into();
 
             let mut closure_bytes = Vec::new();
             closure_ipld
                 .encode(DagCborCodec, &mut closure_bytes)
                 .expect("CBOR Serialization");
 
-            let closure_cid = Cid::try_from(closure_bytes)?
+            let link: Link<Closure> = Closure::try_into(closure)?;
+            let closure_cid = link
                 .to_string_of_base(Base::Base32HexLower)
                 .expect("string CID");
```

Comment:
I have no problem with this being reworked, but could you explain a bit about what the advantage of allocating another binding here is? I don't see `link` used anywhere else 🤔

Reply:
I broke this from your original code, as I was running up the "demo" in its entirety now. Cid (from the multiformat library) will throw an error on `try_into` (when given `closure_bytes`) as it can't decipher the version. This is to get the cid that we need.

Reply:
But, something I'd like to re-explore in how it's derived.

Reply:
This is leveraging the free "into" that comes from TryFrom here in closure.rs:

```rust
impl TryFrom<Closure> for Link<Closure> {
    type Error = anyhow::Error;

    fn try_from(closure: Closure) -> Result<Link<Closure>, Self::Error> {
        let ipld: Ipld = closure.into();
        let bytes = DagCborCodec.encode(&ipld)?;
        Ok(Link::new(Cid::new_v1(
            DagCborCodec.into(),
            Code::Sha3_256.digest(&bytes),
        )))
    }
}
```